File: python_sdk/src/reality_capture/specifications/cs_tiler.py
from typing import Optional
from pydantic import BaseModel, Field
from enum import Enum


class CSTilerInputs(BaseModel):
    scene: str = Field(description="ContextScene reality data id to tile")

class CSObject(Enum):
    CAMERAS = "Cameras"
    TIE_POINTS = "TiePoints"
    ANNOTATIONS = "Annotations"


class CSTilerOptions(BaseModel):
    object_to_tile: Optional[CSObject] = Field(None, description="Object to tile inside the ContextScene.")


class ContextSceneTilerSpecificationsCreate(BaseModel):
    inputs: CSTilerInputs = Field(description="Inputs")
    options: Optional[CSTilerOptions] = Field(None, description="Options")


class ContextSceneTilerSpecifications(BaseModel):
    inputs: CSTilerInputs = Field(description="Inputs")
    output: str = Field(description="Reality Data id of tiled ContextScene")
    options: Optional[CSTilerOptions] = Field(None, description="Options")

File: typescript/packages/reality-capture/src/specifications/cs_tiler.ts
import { z } from "zod";

export const CSTilerInputsSchema = z.object({
  scene: z.string().describe("ContextScene reality data id to tile"),
});
export type CSTilerInputs = z.infer<typeof CSTilerInputsSchema>;

export enum CSObject {
  CAMERAS = "Cameras",
  TIE_POINTS = "TiePoints",
  ANNOTATIONS = "Annotations",
}

export const CSTilerOptionsSchema = z.object({
  objectToTile: z
    .nativeEnum(CSObject)
    .describe("Object to tile inside the ContextScene.")
    .optional(),
});
export type CSTilerOptions = z.infer<typeof CSTilerOptionsSchema>;

export const ContextSceneTilerSpecificationsCreateSchema = z.object({
  inputs: CSTilerInputsSchema.describe("Inputs"),
  options: CSTilerOptionsSchema.describe("Options").optional(),
});
export type ContextSceneTilerSpecificationsCreate = z.infer<
  typeof ContextSceneTilerSpecificationsCreateSchema
>;

export const ContextSceneTilerSpecificationsSchema = z.object({
  inputs: CSTilerInputsSchema.describe("Inputs"),
  output: z.string().describe("Reality Data id of tiled ContextScene"),
  options: CSTilerOptionsSchema.describe("Options").optional(),
});
export type ContextSceneTilerSpecifications = z.infer<
  typeof ContextSceneTilerSpecificationsSchema
>;
